src/main/java/com/epam/aidial/core/storage/BlobStorageUtil.java

```diff
@@ -0,0 +1,77 @@
+package com.epam.aidial.core.storage;
+
+import com.epam.aidial.core.ProxyContext;
+import io.vertx.core.http.impl.MimeMapping;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class BlobStorageUtil {
+
+    private static final String USER_ROOT_DIR_PATTERN = "Users/%s/files/%s";
```

Comment:
Usually any path starts with `/` and may be ended with `/`. Why should we remove leading and trailing path separators every time?

Reply:
some requests provide a filenames like: /file.txt (download)
some provides a folders like: /tmp or /folder1/ (upload)
some may provide a null/empty path

S3 require some patterns to download/list files, for example file path should not start with `/` and query files should end with `/`

This is a common method to "normalize" path and construct correct one
Another place where we need it - metadata (output formatting)

Reply:
I didn't hear about a such restriction on AWS S3 side. So I believe you.